src/InfoBoxes/Content/Other.cpp

```diff
@@ -167,3 +167,44 @@ InfoBoxContentHorizon::Update(InfoBoxData &data)
 
   data.SetCustom();
 }
+
+
+gcc_pure
+static const TCHAR *
+GetGPSStatus(const NMEAInfo &basic)
+{
+  if (!basic.alive)
+    return N_("Disconnected");
+  else if (!basic.location_available)
+    return N_("Fix invalid");
+  else if (!basic.gps_altitude_available)
+    return N_("2D fix");
+  else
+    return N_("3D fix");
```

Comment:
Returning an untranslated translatable string is an interesting micro-optimisation, but that is worth documenting. Please add API documentation to this function, mentioning that `gettext()` on the return value is necessary.

Reply:
i just take this code from existing StatusPanel code, then i presumed it was good.  I don't have experience on translating system in XCSoar...

Reply:
i don't searched longer, cause (as you can see on screenshot), the translation work fine.

Reply:
Documentation patches for old code are also welcome! Sometimes, the best code documentation comes from people who didn't write the code, because only they know that something is not obvious enough.

Reply:
Now I see you copy'n'pasted somebody else's code. That's **bad**! It's bad because we now have two copies of the same code we have to maintain and keep up to date. You should insert one commit which makes the original code available as a library, and then call into that library.

Reply:
the reason i don't do this is to not make big changes. as you probably already understand, i don't produce better code you has seen.  Then a little addition to help community, YES, big changes to (working fine) code of better programmers than me : NO ;-)

Reply:
I appreciate your motivation to improve XCSoar. But if it takes more time for me to clean up after you than it would require me to write the same feature in the first place, then you're not helping. If you have no motivation to learn to produce better code (as I suggested), then we should close this pull request right now and go back to real work. It's up to you. Your post sounded like "but that's all I can do, and I won't learn it". Maybe I misunderstood, and I'll be happy to see my interpretation proven wrong when your new code arrives.

Reply:
I'm actually testing modified code as you propose it to me.  My comment is just to explain that i come for the first time to open-source project.  I try to do by myself the little addition i would like, to not ask to others, to show that i can spend time too, not only profit about other's work. Then i do it, spend lot of time to test, to compile, to learn how to cross compile, and make it run effectively. the next step in my head is to share, not like an egoist... i know it's 1 gram in a ton, but i spend other big big time to understand git, push, pull, branch, fork, and other system. i just present thinks with modesty. i make 10 lines of code, more an arrangement of existing things than genius part of project.  
Sorry to make it not very well. My goal is really to help, give time to project in return of joy to use it, really conscient of lot of work it is from others.
of course i follow advices. i was testing new code when i receive your comment. I finish to test it, then try to understand how to submit again (just upload, in same fork probably ? ...)
If you prefer in future i don't participate cause too low programmer level, i can of course understand (and it is NOT ironic). i can also ask in forum "Please add nbr sat infobox", or keep my modifications.... i want first to respect others working in the project. really.  And one time again, i will thanks all participant for this fantastic work around xcsoar.

Reply:
It's not about what you can do now, and you don't need to apologize or justify or explain imperfect code. It's about whether you're willing to learn and improve.
If you're already busy hacking the code, then welcome aboard, go ahead, don't let me distract you from that. Come back when you have something new to review, or when you need help.

Reply:
i make a file update and new pull request. new code work fine too, but is cleaner. Thx for advices. ;-)